core/new-gui/src/app/workspace/component/property-editor/property-editor.component.ts

```diff
@@ -83,13 +84,36 @@ export class PropertyEditorComponent {
       metadata => { this.operatorSchemaList = metadata.operators; }
     );
 
-    this.autocompleteService.getAutocompleteAPIExecutedStream().subscribe(
-      () => {
+    this.autocompleteService.getOperatorSchemaChangedStream().subscribe(
+      operatorID => {
         if (this.currentOperatorID) {
-          // used the cached data as new schema's initial data
-          this.currentOperatorInitialData = this.cachedFormData;
-          this.currentOperatorSchema = this.autocompleteService.findAutocompletedSchemaForOperator(
-            this.workflowActionService.getTexeraGraph().getOperator(this.currentOperatorID));
+          //  Check if the operator's property are still all valid. If a property is not valid anymore,
+          //   remove the property of the operator in WorkflowGraph.
+          const operator = this.workflowActionService.getTexeraGraph().getOperator(operatorID);
```

Comment:
where are we checking that the operatorID for which the event is generated is the currentOperatorID?